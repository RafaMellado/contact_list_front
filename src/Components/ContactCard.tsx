import { Card } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import { Contact } from "../Services/Interfaces/Contact";

export interface ContactCardProps {
  item: Contact;
  translations: string;
  children?: React.ReactChild;
}

export function ContactCard({
  item,
  translations,
  children,
}: ContactCardProps) {
  const { t } = useTranslation();

  return (
    <Card>
      <Card.Body>
        <>
          <h2>
            {item.givenname} {item.surname}
          </h2>
          <div className="mb-4">
            <small className="text-secondary">
              {t<string>(`${translations}.email`)}
            </small>
            <h5>{item.email}</h5>

            <small className="text-secondary">
              {t<string>(`${translations}.phone`)}
            </small>
            <h5>{item.phone}</h5>
          </div>

          {children}
        </>
      </Card.Body>
    </Card>
  );
}
