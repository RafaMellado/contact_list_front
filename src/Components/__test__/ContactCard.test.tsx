import { ContactCard } from "../ContactCard";
import { act, render } from "@testing-library/react";
import type { ContactCardProps } from "../ContactCard";

describe(`<${ContactCard.name} />`, () => {
  const contactCardProps: ContactCardProps = {
    item: {
      id: 1,
      givenname: "First",
      surname: "Second",
      email: "Test email",
      phone: "Test phone",
      contact_book_id: 1,
    },
    translations: "contactsShow",
  };

  const factoryComponent = (props: ContactCardProps = contactCardProps) =>
    render(<ContactCard {...props} />);

  test("renders correctly the component", () => {
    const component = factoryComponent();

    act(() => {
      expect(component.asFragment()).toMatchSnapshot();
    });
  });
});
