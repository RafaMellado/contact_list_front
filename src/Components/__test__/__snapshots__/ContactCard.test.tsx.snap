// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ContactCard /> renders correctly the component 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card-body"
    >
      <h2>
        First Second
      </h2>
      <div
        class="mb-4"
      >
        <small
          class="text-secondary"
        >
          contactsShow.email
        </small>
        <h5>
          Test email
        </h5>
        <small
          class="text-secondary"
        >
          contactsShow.phone
        </small>
        <h5>
          Test phone
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
